### PR TITLE
WiP: Improving Bug Reporting Workflow

### PR DESCRIPTION
**Pull request by Abbe98**
Aim: to make it easier and clearer to give useful information to developers without running the application from the command line, etc.

 - [x] Logger errors etc in a file.
 - [x] Make this file easily accessible from within the application("view in finder" or similar).
 - [x] Github issue template explaining how to include the log file and media/spreadsheet files.

 - [x] Check for possible file path issues **Windows**.
 - [x] Investigate if we need to clear the log file at any stage.*

\* someone would need to be a very heavy power user and have a lot of bad files or rows, for the log size to be a issue. We could add it later if it becomes a issue or we begin to log a lot more data.

### CODE REVIEW

File: src/pattypan/LogManager.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+public class LogManager {
+  public Logger logger = Logger.getLogger(Main.class.getName());
+
+  public LogManager() {
+    try {
+      Path logFileLocation = Paths.get(System.getProperty("user.home") + "/.PattyPan");
```
Comment (yarl) [inline src/pattypan/LogManager.java]: check `getPropertiesDirectiry()` (yeah, there is a typo...) and `readProperties()` in `Settings.java`

File: src/pattypan/panes/StartPane.java [hunk 2]
```diff
@@ -72,13 +73,17 @@ public GridPane getContent() {
...
+    bugLink.setOnAction(event -> {
+      Util.openUrl("https://github.com/yarl/pattypan/issues/new");
     });
 
     downloadLink.setOnAction(event -> {
       Util.openUrl("https://github.com/yarl/pattypan/releases");
     });
+
+    logFile.setOnAction(event -> {
+      Util.openDirectory(Paths.get(System.getProperty("user.home") + "/.PattyPan"));
```
Comment (yarl) [inline src/pattypan/panes/StartPane.java]: (same comment as in `LogManager.java`)

File: src/pattypan/Settings.java [hunk 3]
```diff
@@ -34,14 +34,13 @@
...
 import java.util.logging.Level;
-import java.util.logging.Logger;
 
 public final class Settings {
 
   private Settings() {};
 
   public static final String NAME = "pattypan";
-  public static final String VERSION = "17.12";
+  public static final String VERSION = "17.05";
```
Comment (yarl) [inline src/pattypan/Settings.java]: please bump to `17.12`, just for diff clarity :)

File: src/pattypan/LogManager.java (hunk 1 above)
Comment (Abbe98) [reply src/pattypan/LogManager.java]: Nice one, I renamed it and moved it to Utilities as its scope has changed... :-)

**Comment by Abbe98** [comment]
On the matter of the log:

I think its good to only have one log file for the end user, for erase of use, if an user goes back to complimenting a bug report or just report something later etc...

To begin with the PattyPan version could be stored in `config.properties` and whenever it is updated the log would be emptied?

**Pull request merged**